VERSIONING.md

```diff
@@ -0,0 +1,103 @@
+"Unshipped.txt" file. Once 1.0.0 is shipped, it'll be moved to "Shipped.txt"
+file.
+
+## Packaging
+
+OpenTelemetry is structured around signals like Traces, Metrics, Logs, Baggage
+etc. OpenTelemetry .NET does not ship separate package per signal. There is a
+single package which includes all the signals. i.e `OpenTelemetry.API` package
+will consist of API components from *all* the signals. `OpenTelemetry` package
+will consist of SDK components from *all* the signals. Instrumentations also
+follow the same model - for example, There will be a single package
+`OpenTelemetry.Instrumentation.AspNetCore` for ASP.NET Core instrumentation,
+which produces Traces, Metrics and handles propagation, instead of separate
+packages for traces and metrics.
+
+## Experimental Signals
+
+### Current approach
+
+Given the fact that all signals are shipped as a single package, the following
+approach is currently used to deal with experimental signals.
+
+Any experimental signal which is shipped as part of a normal package will be
+marked
+[Obsolete](https://docs.microsoft.com/dotnet/api/system.obsoleteattribute).
+(Ideally we need an "Experimental" attribute, but .NET has only built-in support
+for "Obsolete", so we are leveraging it. Using "Obsolete" code results in
+compile time warnings to caution users.) This allows an experimental signal to
+co-exist in the same package as other, non-experimental signals. For example,
+the OpenTelemetry 1.0.0 package will consist of Traces and Metrics. As Metrics
+signal is not ready for stable release, every method dealing with Metrics will
+be marked "Obsolete". Once Metrics signal gets stable, it'll be added to a
+release with minor version bump, say 1.2.0. The "Obsolete" Metric methods can be
+removed with a major version bump.
+
+### Alternate option
+
+An alternate option to deal with experimental signal is to remove the signal
+from the common packages, and ship as separate package. For example,
+OpenTelemetry 1.0.0 will only contain Traces, and there will be a separate
+package OpenTelemetry.Metrics.Experimental, which contains the metric signal.
+Once the signal achieve stable quality, it'll be made part of the main package
+and released as a minor version update. i.e OpenTelemetry 1.2.0 will contain
+Traces and Metrics.
+
+## Examples
+
+Following shows an example on how the `OpenTelemetry` package versioning works:
+
+`OpenTelemetry` 0.7.0-beta1 release : Pre-release, no API guarantees.
+
+`OpenTelemetry` 1.0.0-RC1 release : Pre-release, no API guarantees, but more
+stable than beta.
+
+`OpenTelemetry` 1.0.0 release : Stable release consisting of Traces, Propagators
+and Metrics. Metrics will be marked "Obsolete", to warn users that Metrics is
+not Stable.
+
+`OpenTelemetry` 1.0.1 release : Bug fixes.
+
+`OpenTelemetry` 1.1.0 release : New features added.
+
+`OpenTelemetry` 1.2.0 release : Add metric support. This will be additive changes.
```

Comment:
Would all the methods for metrics previously marked obsolete be removed? Or would that not happen until 2.0.0?

Reply:
Shouldn't we keep it until 2.0 ? So that, if someone took a dependency on it, they can still compile their code without issues when upgrading to 1.2.0.

Reply:
> Or would that not happen until 2.0.0?

That's (more or less) the idea. In OpenTracing we used to remove Deprecated members right in the next release and, well, I remember quite a few complains :)

Reply:
Hmm, yea I see how this may become tricky. If we see a desire to change the name or signatures of methods as we evolve things, then I think that would be mean we'd have to maintain all variations of these methods if we require that an upgrade to 1.2.0 must still compile fine.

Makes me more strongly favor the separate package approach.